### PR TITLE
Text Summarization Model

### LINKED ISSUE #139: Text Summarizing NLP Model
🔍 **Problem Description**:
It helps users to summarize huge texts

🧠 **Model Description**:
It will be a text summarization NLP model where user and type their huge paragraphs and the model will summarize it 

⏲️ **Estimated Time for Completion**:
Within 1 day
🎯 **Expected Outcome**:
People do face issues to summarize huge texts or students do face issues to learn huge texts. this model will help users to summarize those texts and use it according to their need, for example a learner can easily learn faster  and understand the summarized text instead of going through huge paragraphs 
📄 **Additional Context**:

**To be Mentioned while taking the issue**:
- What is your participant role? GSSOC-24 EXTD , Hactoberfest

**Note:**
- Please review the project documentation and ensure your code aligns with the project structure.
- Please ensure that either the `predict.py` file includes a properly implemented `model_details()` function or the notebook contains this function to print a detailed model report. The model will not be accepted without this function in place, as it is essential for generating the necessary model details.
- Prefer using a new branch to resolve the issue, as it helps keep the main branch stable and makes it easier to manage and review your changes.
- Strictly use the pull request template provided in the repository to create a pull request.

### PR DESCRIPTION
**Pull request by Peart-Guy**
### Description
I added a Text Summarizer Model 
Name: Ankan
Github: Peart-Guy
Participant role: GSSOC-24 EXTD , Hacktoberfest

### Issue Resolved
#139 

### Changes Made
I added a new model which summarizes long texts that enhances understanding and speed up learning—perfect for students and professionals alike.

### Screenshots or Videos

https://github.com/user-attachments/assets/5b00cfe0-2dff-4790-b88b-7cc1647f0460

### Checklist

- [x] My code follows the current [project structure](https://github.com/yashasvini121/predictive-calc/blob/master/docs/project-structure.md)
- [x] I have thoroughly reviewed and updated the `requirements.txt` file to include any new packages
- [ ] The `predict.py` file includes a properly implemented `model_details()` function, or the notebook contains this function to print a detailed model report. **The model will not be accepted without this function**, as it is essential for generating the necessary model details.
- [x] I have added relevant tests (if necessary).
- [x] I have added comments in the code where needed.
- [x] This PR is submitted under **Hacktoberfest**.
- [x] This PR is submitted under **GirlScript Summer of Code (GSSoC-Extd)**.

### CODE REVIEW

**Comment by Peart-Guy** [comment]
@yashasvini121  I have done my work. Kindly check it out and give GSSOC-24 EXTD, Hacktoberfest and Level 3. It will be an honor

**Review by yashasvini121 — CHANGES_REQUESTED** [review]
Your work in not in the correct project structure. Please refer docs/

**Comment by Peart-Guy** [comment]
> Your work in not in the correct project structure. Please refer docs/

Since my program is a NLP model so it is done in a single file and it does not need any separate code file to train or a separate code file to run or a separate file of data sets @yashasvini121

File: models/text_sumarization/predict.py
Comment (yashasvini121) [inline models/text_sumarization/predict.py:1]: The `predict.py` and `pages/text_summarizer.py` files contain identical code. Please remove the Streamlit-related code from `predict.py`. You don't need to manually create the page—use the `page_handler` class to generate it. Refer `docs/`

**Review by yashasvini121 — CHANGES_REQUESTED** [review]
Your final submission should include the following:  
1. An independent `predict.py` file using **Streamlit**.  
2. Your page added to the `pages.json` configuration.  
3. Your form configurations defined in `form_config`.  
4. Utilization of the `page_handler` class to generate your page.  

Check docs in case, I've missed anything, but these are the primary requirements.

**Comment by yashasvini121** [comment]
Feel free to reach out if you have any questions regarding the expected changes.

**Comment by Peart-Guy** [comment]
> Feel free to reach out if you have any questions regarding the expected changes.

@yashasvini121 
Since my project comes with the built in UI so it is completed in one single file. My project does not need separate predict.py file or separate UI file. It will completely run by the Text Summarizer.py in the pages folder 


I hope I was able to explain everything clearly.  Would you like any further adjustments?

**Comment by yashasvini121** [comment]
> > Feel free to reach out if you have any questions regarding the expected changes.
> 
> @yashasvini121 Since my project comes with the built in UI so it is completed in one single file. My project does not need separate predict.py file or separate UI file. It will completely run by the Text Summarizer.py in the pages folder
> 
> I hope I was able to explain everything clearly. Would you like any further adjustments?

Your project works fine, but please note that the PR checklist specifies that the required project structure must be followed. If not, your PR will only receive a Level 1 rating.

**Comment by Peart-Guy** [comment]
> > > Feel free to reach out if you have any questions regarding the expected changes.
> > 
> > 
> > @yashasvini121 Since my project comes with the built in UI so it is completed in one single file. My project does not need separate predict.py file or separate UI file. It will completely run by the Text Summarizer.py in the pages folder
> > I hope I was able to explain everything clearly. Would you like any further adjustments?
> 
> Your project works fine, but please note that the PR checklist specifies that the required project structure must be followed. If not, your PR will only receive a Level 1 rating.



@yashasvini121 
But I don't have extra files to add ,to maintain the project structure.
Kindly understand that  my project is designed to be simple and efficient, requiring only a single file to function perfectly. This streamlined approach reflects my preference for minimalistic programming, ensuring both ease of maintenance and clarity in structure.
I trust that I have clearly explained the issue, and I am confident that you will understand my perspective.
Thank You

**Comment by Peart-Guy** [comment]
> > > > Feel free to reach out if you have any questions regarding the expected changes.
> > > 
> > > 
> > > @yashasvini121 Since my project comes with the built in UI so it is completed in one single file. My project does not need separate predict.py file or separate UI file. It will completely run by the Text Summarizer.py in the pages folder
> > > I hope I was able to explain everything clearly. Would you like any further adjustments?
> > 
> > 
> > Your project works fine, but please note that the PR checklist specifies that the required project structure must be followed. If not, your PR will only receive a Level 1 rating.
> 
> @yashasvini121 But I don't have extra files to add ,to maintain the project structure. Kindly understand that my project is designed to be simple and efficient, requiring only a single file to function perfectly. This streamlined approach reflects my preference for minimalistic programming, ensuring both ease of maintenance and clarity in structure. I trust that I have clearly explained the issue, and I am confident that you will understand my perspective. Thank You



@yashasvini121 I hope you understood. Will be waiting for your reply
Thank You

**Review by yashasvini121 — APPROVED** [review]

**Comment by Peart-Guy** [comment]
@yashasvini121  Thank You for Understanding. It was pleasure working with you.

**Pull request merged**